### PR TITLE
Display tap headers in dashboard

### LINKED ISSUE #3612: Display tap headers in dashboard view
#3434 and #3410 added request and response headers to `linkerd tap` when using the `-o json` flag.

```
linkerd tap ns/linkerd -ojson
```

These headers now be displayed in the dashboard. We can start to incorporate the headers into the Tap view by adding them to the detail modal that shows when clicking the `ˇ` icon on the left of each returned row:

(below screenshots from `tap?namespace=linkerd&resource=namespace%2Flinkerd`)

Screen Shot 2019-10-21 at 12 57 08 PM

We can add the request and response headers to the modal at these 2 locations:

Screen Shot 2019-10-21 at 1 01 16 PM

### PR DESCRIPTION
**Pull request by cynthia-sg**
Tap headers can now be displayed in the dashboard from the tap view.

tap-headers-2

Closes #3612

Signed-off-by: Cintia Sanchez Garcia <cynthiasg@icloud.com>

### CODE REVIEW

File: web/app/js/components/TapEventTable.jsx [hunk 1]
```diff
@@ -126,6 +126,67 @@ const formatTapLatency = str => {
...
+};
+
+const HeadersContentBase = ({headers, classes}) => {
+  return (
+    <React.Fragment>
+      {headers.map(header => {
+        return (
+          <Typography
+            key={`${header.name}_${header.valueStr}`}
+            component="span"
```
[the remark this answers is not in the document]
Comment (cynthia-sg) [reply web/app/js/components/TapEventTable.jsx]: Not, it doesn't offer any advantage. I tried to use different components: `li`, `p`... to test different styles for the headers and I decided to leave the default one, but I forgot to delete the `prop`. Thanks

File: web/app/js/components/ExpandableTable.jsx [hunk 2]
```diff
@@ -105,6 +105,7 @@ class ExpandableTable extends React.Component {
         </Table>
         <Dialog
           maxWidth="md"
+          fullWidth
```
[the remark this answers is not in the document]
Comment (cynthia-sg) [reply web/app/js/components/ExpandableTable.jsx]: Yes, I think it looks nicer a wider dialog, but.. please, let me know your thoughts.

**Comment by cynthia-sg** [comment]
Hi @scottcarol,

I have removed the checkbox from `Tap filters` and I'm adding `extract: true` to query. About your UI notes:

1. Do you mean to use `scroll="body"` for `Dialog`? I have applied this property. If you are happy with this, please let me know. If you don't like this solution and the default`scroll="paper"` neither, we can find another solution... but then we'll have to update the styles from different Material UI components.
2. Fixed (issue is related with https://material-ui.com/components/grid/#limitations)
3. Sorry, I'm not able to reproduce this issue. Could you please give me more details: browser, version...?

Thanks

File: web/app/js/components/TapEventTable.jsx [hunk 3]
```diff
@@ -126,6 +126,65 @@ const formatTapLatency = str => {
   return formatLatencySec(str.replace("s", ""));
 };
 
+const headersStyles = {
```
Comment (scottcarol) [inline web/app/js/components/TapEventTable.jsx]: I think the UI and display looks great. In reviewing the whole `TapEventTable` component, I am concerned that the component is getting to be too long and complex. Would you be able to break the headers content into its own component file and call it from the `TapEventTable` component?

Comment (cynthia-sg) [reply web/app/js/components/TapEventTable.jsx]: Done 👍

File: web/app/js/components/TapEventHeadersTable.jsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  }
+
+  return (
+    <ListItem disableGutters>
+      <ListItemText
+        primary={title}
+        secondary={"headers" in value ? <HeadersContentDisplay headers={value.headers} /> : "-"} />
+    </ListItem>
+  );
+};
```
Comment (scottcarol) [inline web/app/js/components/TapEventHeadersTable.jsx]: Can you fix the newline warning here?

**Pull request merged**